mobile/src/containers/onboarding/SelectServer.jsx

```diff
@@ -1,30 +1,73 @@
 import React from 'react'
 import { connect } from 'react-redux'
-import classnames from 'classnames'
+import classNames from 'classnames'
 
 import { translate } from '../../../../src/lib/I18n'
 
 import { registerDevice, setUrl } from '../../actions/settings'
 import styles from '../../styles/onboarding'
 
-export const SelectServer = ({selectServer, t, updateServerUrl, serverUrl, error}) =>
+export const SelectServer = ({t, goBack, selectServer, updateServerUrl, serverUrl, error, authorized}) =>
 (
-  <div className={classnames(styles['wizard'])}>
-    <div className={classnames(styles['wizard-main'])}>
-      <p>{t('mobile.onboarding.server_selection.cozy_address')}</p>
-      <input type='url' placeholder={t('mobile.onboarding.server_selection.cozy_address_placeholder')} onChange={updateServerUrl} value={serverUrl} />
-      <p>{t('mobile.onboarding.server_selection.description')}</p>
-      {error && <p style={{color: 'red'}}>{t(error)}</p>}
+  <div className={classNames(styles['wizard'], styles['select-server'])}>
+    <header className={styles['wizard-header']}>
+      <a
+        className={styles['close-button']}
+        onClick={goBack}
+      />
+    </header>
+    <div className={styles['wizard-main']}>
+      <div
+        className={error
+          ? classNames(styles['logo-wrapper'], styles['error'])
+          : styles['logo-wrapper']}
+      >
+        <div className={styles['cozy-logo-white']} />
+      </div>
+      <input
+        type='url'
+        className={error
+          ? classNames(styles['input'], styles['error'])
+          : styles['input']}
+        placeholder={t('mobile.onboarding.server_selection.cozy_address_placeholder')}
+        onChange={updateServerUrl}
+        value={serverUrl}
+      />
+      {!error &&
+        <p className={styles['description']}>
+          {t('mobile.onboarding.server_selection.description')}
+        </p>
+      }
+      {error &&
+        <p className={styles['description']} style={{color: 'red'}}>
+          {t(error)}
+        </p>
+      }
     </div>
-    <button role='button' className={classnames('coz-btn coz-btn--regular', styles['wizard-button'])} onClick={selectServer}>{t('mobile.onboarding.server_selection.button')}</button>
+    <footer className={styles['wizard-footer']}>
+      <button
+        role='button'
+        className={'coz-btn coz-btn--regular'}
+        onClick={() => selectServer(serverUrl, authorized)}
+        disabled={error}
+      >
+        {t('mobile.onboarding.server_selection.button')}
+      </button>
+    </footer>
   </div>
 )
 
 const mapDispatchToProps = (dispatch, ownProps) => ({
-  selectServer: () => {
-    dispatch(registerDevice()).then(() => {
-      ownProps.nextStep()
-    })
+  goBack: () => {
+    ownProps.previousStep()
+  },
+  selectServer: (serverUrl, authorized) => {
+    if (!serverUrl) return
+    dispatch(registerDevice())
+      .then(() => {
+        if (authorized) ownProps.nextStep()
```

Comment:
Agree with you, that probably was a previous fix. I merged the PR you did to fix that. Thanks 👍